src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -273,14 +288,29 @@ protected async Task<IActionResult> ExecuteQuery<TPostBodyResource>(TResource re
 
             var dataProviders = new DataProviders.DataProviders((DataProviderContext)HttpContext.Items[HostConstants.DataProviderContextKey]);
 
-            foreach (var p in _gistCache.GetAllReferences(runtimeContext))
+            if(loadGistFromRepo)
             {
-                if (!jsonBody.References.ContainsKey(p.Key))
+                List<string> gistReferences = DetectorParser.GetLoadDirectiveNames(jsonBody.Script);
+                foreach (string gist in gistReferences)
                 {
-                    // Add latest version to references
-                    jsonBody.References.Add(p);
+                    if(!jsonBody.References.ContainsKey(gist))
+                    {
+                        object gistContent = await devopsClient.GetFileContentAsync($"{gist}/{gist}.csx", resource.ResourceUri, HttpContext.Request.Headers[HeaderConstants.RequestIdHeaderName]);
```

Comment:
Quick question here - For every single ExecuteQuery(i.e. Compilation request), we are getting each gist file code from devops repo, do you forsee us hitting throttling limits on devops pretty frequently? 
Maybe we can consider caching the gist content in memory, however the question would be how to invalidate the cache. Something to think about.

Reply:
Yes Shekhar,  I also thought of implementing a cache at first. I also felt there would be a situation where gist code in the cache might become out of sync with latest code in repo (In case it gets updated via separate PR). For the first iteration, I am taking this approach and will be improving based on usage and metrics (Number of times compilation is called)